streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -234,10 +256,10 @@ private ClientResponse sendRequest(final ClientRequest clientRequest) {
     /**
      * Fetch the metadata for all topics
      */
-    public Collection<MetadataResponse.TopicMetadata> fetchTopicsMetadata() {
+    public MetadataResponse fetchMetadata() {
 
         final ClientRequest clientRequest = kafkaClient.newClientRequest(
-            getBrokerId(),
+            getAnyReadyBrokerId(),
```

Comment:
The case where you should fetch the metadata from the controller is if you expect the metadata response to include the topic that was just created. For example, if you create a topic, it returns a success and _then_ you ask any broker for the metadata, it may be missing the topic that was just created (since metadata updates are asynchronous). I haven't checked the code to understand if this matters or not.